### PR TITLE
feat: #1471 Handle email sending error

### PR DESCRIPTION
**Pull request by ianliuwk1019**
ref: #1471 
- email sending status enum defined to be used for email sending to external service.
- granting user and granting delegated admin endpoints return response changed to incorporate email sending status.
- adjust router and service due to response schema change.
- add 'with_client_number' email param to schema and its conditonal expression for email service
- set email sending status depending on sending result to the response.
- generate api client.
- adjust frontend to show email sending failure message when email sending failed.

### CODE REVIEW

File: server/admin_management/api/app/schemas.py [hunk 1]
```diff
@@ -339,4 +344,5 @@ class GCNotifyGrantDelegatedAdminEmailParam(BaseModel):
     last_name: Annotated[str, StringConstraints(max_length=20)]
     role_list_string: Annotated[str, StringConstraints(max_length=200)]
     application_team_contact_email: Optional[EmailStr] = None
+    with_client_number: Literal['yes', 'no']
```
Comment (ianliuwk1019) [inline server/admin_management/api/app/schemas.py:347]: This is new GC Notify's variable for conditional display (it needs value as 'yes' or 'no').

File: frontend/src/components/grantaccess/form/UserNameInput.vue [hunk 2]
```diff
@@ -72,7 +72,7 @@ const verifyUserId = async () => {
         }
     } finally {
         if (verifiedUserIdentity.value?.found) {
-            emit('setVerifyResult', true, verifiedUserIdentity.value.guid);
+            emit('setVerifyResult', true, verifiedUserIdentity.value.guid, verifiedUserIdentity.value.email);
```
Comment (ianliuwk1019) [inline frontend/src/components/grantaccess/form/UserNameInput.vue:75]: emit searched user 'email'. This is then used for displaying "failed to send email to" error message based on mockup.

File: server/admin_management/api/app/routers/router_access_control_privilege.py [hunk 3]
```diff
@@ -32,7 +27,7 @@
 
 @router.post(
     "",
-    response_model=List[schemas.FamAccessControlPrivilegeCreateResponse],
+    response_model=schemas.FamAccessControlPrivilegeResponse,
```
Comment (MCatherine1994) [inline server/admin_management/api/app/routers/router_access_control_privilege.py:30]: Hi Ian, I think you already imported the FamAccessControlPrivilegeResponse from schemas, so maybe no need to call it schemas.FamAccessControlPrivilegeResponse

File: server/admin_management/api/app/schemas.py (hunk 1 above)
Comment (MCatherine1994) [reply server/admin_management/api/app/schemas.py:347]: thanks Ian, later on once we support grant multiple roles might need to look at this again, good for now

File: frontend/src/components/grantaccess/GrantDelegatedAdmin.vue [hunk 4]
```diff
@@ -136,6 +139,12 @@ const confirmSubmit = async () => {
                 errorList.push(forestClientNumber ?? '');
             }
         });
+        if (returnResponse.data.email_sending_status == EmailSendingStatus.SentToEmailServiceFailure) {
+            setNotificationMsg(
```
Comment (MCatherine1994) [inline frontend/src/components/grantaccess/GrantDelegatedAdmin.vue:143]: Did we test the situation when email failed to sent, and there is also an error when grant duplication access? I wonder if the error message will override each other...

File: server/admin_management/api/app/services/access_control_privilege_service.py [hunk 5]
```diff
@@ -234,6 +232,7 @@ def send_email_notification(
                 return
```
Comment (MCatherine1994) [inline server/admin_management/api/app/services/access_control_privilege_service.py:232]: do we want to return the status not required in this case?

File: server/backend/api/app/routers/router_user_role_assignment.py [hunk 6]
```diff
@@ -24,7 +25,7 @@
 
 @router.post(
     "",
-    response_model=List[schemas.FamUserRoleAssignmentCreateResponse],
+    response_model=schemas.FamUserRoleAssignmentResponse,
```
Comment (MCatherine1994) [inline server/backend/api/app/routers/router_user_role_assignment.py:28]: Hi Ian, same here, maybe you can use this schema directly as it's imported

File: server/admin_management/api/app/routers/router_access_control_privilege.py (hunk 3 above)
Comment (ianliuwk1019) [reply server/admin_management/api/app/routers/router_access_control_privilege.py:30]: Right, maybe I was confused that time from IDE, will remove it.

File: frontend/src/components/grantaccess/GrantDelegatedAdmin.vue (hunk 4 above)
Comment (ianliuwk1019) [reply frontend/src/components/grantaccess/GrantDelegatedAdmin.vue:143]: Hm... I think you are right, I didn't corelated the case. For too long, I think I am losing frontend sense even when I saw previous lines "errorList" and was confused. Need to look into it. Thank you so much!!

File: server/admin_management/api/app/services/access_control_privilege_service.py (hunk 5 above)
Comment (ianliuwk1019) [reply server/admin_management/api/app/services/access_control_privilege_service.py:232]: That should be good, the default is not_required in the variable.

File: server/backend/api/app/routers/router_user_role_assignment.py (hunk 6 above)
Comment (ianliuwk1019) [reply server/backend/api/app/routers/router_user_role_assignment.py]: Removed.

File: frontend/src/components/grantaccess/GrantDelegatedAdmin.vue (hunk 4 above)
Comment (ianliuwk1019) [reply frontend/src/components/grantaccess/GrantDelegatedAdmin.vue:143]: Ticket #1532 is added for dealing with this message overriden issue.

**Comment by MCatherine1994** [comment]
thanks so much Ian!

**Review by MCatherine1994 — APPROVED** [review]

**Pull request merged**